### ISSUE TITLE
Bug in type casting for wires

### ISSUE THREAD

**Issue by Francesco-Benfenati** [PennyLaneAI/pennylane#830]
#### Issue description

Evaluation of operators fails when using plugins because circuits/gates are invoked with a pennylane wires object instead of a number or a slice.

This issue can be easily reproduced running the vqe tutorial https://pennylane.ai/qml/demos/tutorial_vqe.html and changing the device at line 120 to, for example,
`dev = qml.device('qulacs.simulator', wires=nr_qubits)`

* *Expected behavior:* (What you expect to happen)
No errors given

* *Actual behavior:* (What actually happens)
Various errors are given depending on the device used

* *Reproduces how often:* (What percentage of the time does it reproduce?)
Every time

* *System information:* (post the output of `import pennylane as qml; qml.about()`)

Name: PennyLane
Version: 0.11.0
Summary: PennyLane is a Python quantum machine learning library by Xanadu Inc.
Home-page: https://github.com/XanaduAI/pennylane
Author: None
Author-email: None
License: Apache License 2.0
Location: /root/anaconda3/envs/portfolio/lib/python3.7/site-packages
Requires: networkx, scipy, autograd, appdirs, semantic-version, toml, numpy
Required-by: pennylane-qulacs, PennyLane-qiskit, PennyLane-Qchem, PennyLane-PQ, PennyLane-Forest
Platform info:           Linux-4.4.0-19041-Microsoft-x86_64-with-debian-buster-sid
Python version:          3.7.9
Numpy version:           1.19.2
Scipy version:           1.5.2
Installed devices:
- default.gaussian (PennyLane-0.11.0)
- default.qubit (PennyLane-0.11.0)
- default.qubit.autograd (PennyLane-0.11.0)
- default.qubit.tf (PennyLane-0.11.0)
- default.tensor (PennyLane-0.11.0)
- default.tensor.tf (PennyLane-0.11.0)
- qulacs.simulator (pennylane-qulacs-0.0.4)
- qiskit.aer (PennyLane-qiskit-0.8.2)
- qiskit.basicaer (PennyLane-qiskit-0.8.2)
- qiskit.ibmq (PennyLane-qiskit-0.8.2)
- projectq.classical (PennyLane-PQ-0.11.0)
- projectq.ibm (PennyLane-PQ-0.11.0)
- projectq.simulator (PennyLane-PQ-0.11.0)
- forest.numpy_wavefunction (PennyLane-Forest-0.8.0)
- forest.qpu (PennyLane-Forest-0.8.0)
- forest.qvm (PennyLane-Forest-0.8.0)
- forest.wavefunction (PennyLane-Forest-0.8.0)

#### Source code and tracebacks

For example using the qiskit plugin:
`CircuitError: 'expected integer or slice index into register'`

When using forest:
`TypeError: int() argument must be a string, a bytes-like object or a number, not 'Wires'`

When using qulacs:
```
TypeError: RZ(): incompatible function arguments. The following argument types are supported:
    1. (index: int, angle: float) -> qulacs.QuantumGateBase
Invoked with: <Wires = [0]>, 5.541933890639854

```

#### Additional information
Might be related to issue 456 https://github.com/PennyLaneAI/pennylane/issues/456

**Comment by antalszava** [comment]
Hi @Francesco-Benfenati, thank you so much for reporting this! We're looking into it, based on the trace it would be related to a refactor that was made connected to how wires are represented internally. Will be getting back to you on it. :slightly_smiling_face:

**Comment by thisac** [comment]
Hi @Francesco-Benfenati. It seems like you're using an old version of PennyLane-Qulacs (`qulacs.simulator (pennylane-qulacs-0.0.4)` in the output you attached). The newest version should be `pennylane-qulacs-0.11.0`, which you can install/update to by using `pip install pennylane-qulacs`. Could you try updating your version of PennyLane-Qulacs and see if that solves your issue?

**Comment by Francesco-Benfenati** [comment]
Hi @thisac, thanks! You're right, with the latest version of PennyLane-Qulacs the issue is solved, for the qulacs plugin. Cirq is also working. I also found that there is a newer version 0.11.0 of the pennylane-qiskit plugin that is working, my bad! I don't know why I had an old version, because I had just installed them a few days ago using pip.

The problem remains for the pennylane-forest plugin because the latest version on PyPi is 0.9.0, but I see that on GitHub the latest version is 0.11.0. I think PyPi needs to be updated. The pennylane-qsharp plugin also seems outdated, so it might have the same problem.

Thank you!
